test/Dialect/Torch/canonicalize.mlir

```diff
@@ -3,25 +3,63 @@
 // CHECK-LABEL:   func @torch.aten.__is__
 // CHECK:           %[[FALSE:.*]] = basicpy.bool_constant false
 // CHECK:           return %[[FALSE]] : !basicpy.BoolType
-func @torch.aten.__is__(%arg0: !basicpy.ListType, %arg1: !basicpy.NoneType) -> !basicpy.BoolType{
-  %0 = torch.aten.__is__ %arg0, %arg1 : !basicpy.ListType, !basicpy.NoneType -> !basicpy.BoolType
+func @torch.aten.__is__(%arg0: !torch.list<i64>, %arg1: !basicpy.NoneType) -> !basicpy.BoolType{
+  %0 = torch.aten.__is__ %arg0, %arg1 : !torch.list<i64>, !basicpy.NoneType -> !basicpy.BoolType
   return %0 : !basicpy.BoolType
 }
 // CHECK-LABEL:   func @torch.aten.size(
 // CHECK:           %[[CM1:.*]] = constant -1 : i64
 // CHECK:           %[[C3:.*]] = constant 3 : i64
-// CHECK:           %[[RET:.*]] = basicpy.build_list %[[CM1]], %[[C3]] : (i64, i64) -> !basicpy.ListType
-// CHECK:           return %[[RET]] : !basicpy.ListType
-func @torch.aten.size(%arg0: tensor<?x3xf32>) -> !basicpy.ListType {
-  %0 = torch.aten.size %arg0 : tensor<?x3xf32> -> !basicpy.ListType
-  return %0 : !basicpy.ListType
+// CHECK:           %[[RET:.*]] = torch.prim.ListConstruct %[[CM1]], %[[C3]] : (i64, i64) -> !torch.list<i64>
+// CHECK:           return %[[RET]] : !torch.list<i64>
+func @torch.aten.size(%arg0: tensor<?x3xf32>) -> !torch.list<i64> {
+  %0 = torch.aten.size %arg0 : tensor<?x3xf32> -> !torch.list<i64>
+  return %0 : !torch.list<i64>
 }
 
 // CHECK-LABEL:   func @torch.aten.len.t(
 // CHECK:           %[[LENGTH:.*]] = constant 2 : i64
 // CHECK:           return %[[LENGTH]] : i64
 func @torch.aten.len.t(%arg0: i64) -> i64 {
-  %0 = basicpy.build_list %arg0, %arg0 : (i64, i64) -> !basicpy.ListType
-  %1 = torch.aten.len.t %0 : !basicpy.ListType -> i64
+  %0 = torch.prim.ListConstruct %arg0, %arg0 : (i64, i64) -> !torch.list<i64>
+  %1 = torch.aten.len.t %0 : !torch.list<i64> -> i64
   return %1 : i64
 }
+
+// CHECK-LABEL:   func @torch.aten.__getitem__.t(
+// CHECK:           %[[C5:.*]] = constant 5 : i64
+// CHECK:           return %[[C5]] : i64
+func @torch.aten.__getitem__.t() -> i64 {
+    %c4_i64 = constant 4 : i64
+    %c5_i64 = constant 5 : i64
+    %c1_i64 = constant 1 : i64
+    %0 = torch.prim.ListConstruct %c4_i64, %c5_i64 : (i64, i64) -> !torch.list<i64>
+    %1 = torch.aten.__getitem__.t %0, %c1_i64 : !torch.list<i64>, i64 -> i64
+    return %1 : i64
+}
+
+// Not canonicalized because of passed in index
+// CHECK-LABEL:   func @torch.aten.__getitem__.t_no_change_test0(
+// CHECK:           %[[C4:.*]] = constant 4 : i64
+// CHECK:           %[[C5:.*]] = constant 5 : i64
+// CHECK:           %[[LIST:.*]] = torch.prim.ListConstruct %[[C4]], %[[C5]] : (i64, i64) -> !torch.list<i64>
+// CHECK:           %[[ITEM:.*]] = torch.aten.__getitem__.t %[[LIST]], %arg0 : !torch.list<i64>, i64 -> i64
+// CHECK:           return %[[ITEM]] : i64
+func @torch.aten.__getitem__.t_no_change_test0(%arg0: i64) -> i64 {
```

Comment:
nit: I would typically do `$no_change_test0` (because `_` is already heavily used as a separator so it is confusing)